calcom/cal.com
# ci: mock stripe for e2e

## What does this PR do?

- Centralizes Stripe configuration and usage across the application
- Adds support for Stripe mock server in development environment
- Updates Stripe API version to "2020-08-27"

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?

1. Set up the Stripe mock server:
   - Run `yarn dx` in the `packages/features/ee` directory to start the Stripe mock server
   - Ensure `STRIPE_MOCK_HOST` and `STRIPE_MOCK_PORT` environment variables are set

2. Test Stripe-related functionality:
   - Create a new team and verify the payment process
   - Test subscription management
   - Verify webhook handling for Stripe events

3. Run existing tests to ensure no regressions

## Checklist

- I have read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code follows the style guidelines of this project
- I have commented my code, particularly in hard-to-understand areas
- I have checked if my changes generate no new warnings
